hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -46,11 +47,16 @@ default boolean supportsSteppedRun() {
     }
 
     /**
-     * Injects extra step for an operation which
-     * is modeled as a sequence of {@link Step}
-     * @return new step to be injected before an operation is finalized.
+     * Appends extra steps to the head of an operation's
+     * existing step chain in the provided order.
+     * <p>
+     * So after append, your 1st step from this new list
+     * will be the first step of the existing step chain,
+     * 2nd will be the 2nd and so on.
+     *
+     * @return list of steps to be appended to head of existing step chain
      */
-    default Step newInjectedStep() {
+    default List<Step> headSteps() {
```

Comment:
For local indexes PR it won't help because `InternalIndex` is not an instance of `Storage`. There is no link from Storage to the associated `InternalIndex`(s)

Reply:
can't you access it over map-container?

In `TieredStorageImpl`, `getMapContainer().getStepAwareStorages(partitionId);` can be called to create `headSteps`?

Reply:
Do we have a link to the MapContainer from TieredStorageImpl? It seems not now.

Reply:
i did it here: https://github.com/hazelcast/hazelcast-enterprise/pull/6454
upon merge it can be used.